File: 2015/D01/main.py
def main():
    fin = open("input.in", "r")
    line = fin.read()

    print("Part 1: " + str(part1(line)))
    print("Part 2: " + str(part2(line)))

def part1(line):
    floor = 0

    for x in line:
        if (x == "("):
            floor += 1
        if (x == ")"):
            floor -= 1

    return floor

def part2(line):
    floor = 0

    for i, x in enumerate(line):
        if (x == "("):
            floor += 1
        if (x == ")"):
            floor -= 1
        if (floor == -1):
            return i+1

    return -1

main()

File: 2015/D01/main.rs
use std::fs;

fn main() {
    let fin = fs::read_to_string("input.in").expect("Input Error");

    println!("Part 1: {}", part1(&fin));
    println!("Part 2: {}", part2(&fin));
}

fn part1(fin: &String) -> String {
    let mut floor: i32 = 0;

    for c in fin.chars() {
        if c == '(' {
            floor += 1;
        }
        if c == ')' {
            floor -= 1;
        }
    }

    floor.to_string();
}

fn part2(fin: &String) -> String {
    let mut floor: i32 = 0;

    for (i, c) in fin.chars().enumerate() {
        if c == '(' {
            floor += 1;
        }
        if c == ')' {
            floor -= 1;
        }
        if floor == -1 {
            return (i + 1).to_string();
        }
    }

    0.to_string();
}
